chromadb/segment/__init__.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from typing import TypedDict, Union
+import numpy.typing as npt
+
+IDSeq = Sequence[str]
+Vector = Sequence[Union[int, float]]
+
+
+class Query(TypedDict):
+    """A KNN/ANN query to the segment"""
+    vector: Vector
+    k: int = 10
+    approximate: bool = True
+    allowed_ids: Optional[IDSeq] = None
+
+
+class Reader(ABC):
+    """Read part of Segment Driver interface, used to retrieve
+    vectors, and to perform KNN/ANN searches."""
+
+
+    @abstractmethod
+    def get(self, ids: Optional[IDSeq]) -> tuple[IDSeq, npt.NDArray]:
+        """Get embeddings from the segment. If no IDs are provided,
+        all embeddings are returned."""
+        pass
+
+    @abstractmethod
+    def search(self, queries: Sequence[Query]) -> Sequence[IDSeq]:
```

Comment:
nit: since we use query at the top-level interface, can we use it here? Or are we changing to search (which I would be OK with)

Reply:
Fine with query, will switch.